### PR TITLE
feat: Gateway configure RoutingFees and number of route hints

### PR DESCRIPTION
**Pull request by m1sterc001guy**
Fixes: https://github.com/fedimint/fedimint/issues/3167

Allows the gateway operator to configure their routing fees and the number of route hints at run time.

### CODE REVIEW

File: gateway/ln-gateway/src/lib.rs [hunk 1]
```diff
@@ -134,6 +134,26 @@ pub struct GatewayOpts {
...
+impl GatewayOpts {
+    fn to_gateway_parameters(&self) -> GatewayParameters {
+        // TODO: Consume self
+        GatewayParameters {
+            listen: self.listen,
+            password: self.password.clone()
+        }
+    }
+}
+*/
```
Comment (elsirion) [inline gateway/ln-gateway/src/lib.rs]: Can we remove this?

**Review by elsirion — DISMISSED** [review]
Looks reasonable, maybe someone with more context about the GW wants to do a more thorough review.

File: gateway/ln-gateway/src/lib.rs [hunk 2]
```diff
@@ -134,6 +134,26 @@ pub struct GatewayOpts {
     pub num_route_hints: Option<usize>,
 }
 
+/*
```
Comment (okjodom) [inline gateway/ln-gateway/src/lib.rs]: nit, commented code?

Comment (m1sterc001guy) [reply gateway/ln-gateway/src/lib.rs]: Fixed in a later commit , perhaps I should squash those two together

File: gateway/ln-gateway/src/lib.rs [hunk 3]
```diff
@@ -500,10 +513,15 @@ impl Gateway {
             lightning_alias,
         } = self.state.read().await.clone()
         {
+            let gateway_config = self
+                .get_gateway_configuration()
+                .await
+                .expect("Gateway configuration should be set");
```
Comment (okjodom) [inline gateway/ln-gateway/src/lib.rs:533]: should we gracefully exit the api call instead?

Comment (m1sterc001guy) [reply gateway/ln-gateway/src/lib.rs:533]: It should be guaranteed to exist at this point. `get_gateway_configuration` should only return `None` if we're in the `Configuring` state. I'll add a comment

**Review by okjodom — DISMISSED** [review]

**Review by okjodom — APPROVED** [review]

**Pull request merged**